pytorch_lightning/trainer/trainer.py

```diff
@@ -61,56 +66,56 @@ class Trainer(TrainerIOMixin,
 
     def __init__(
             self,
-            logger=True,
-            checkpoint_callback=True,
-            early_stop_callback=None,
-            default_save_path=None,
-            gradient_clip_val=0,
+            logger: Union[LightningLoggerBase, bool] = True,
```

Comment:
I guess that all with default value are Optional

Reply:
Optional simply means it can be None. 
See https://docs.python.org/3/library/typing.html#typing.Optional

Reply:
I think Optional is used for arguments that can be explicitly None. 
See https://docs.python.org/3/library/typing.html#typing.Optional